plugins/CoreHome/vue/src/externalLink.ts

```diff
@@ -0,0 +1,106 @@
+/*!
+ * Matomo - free/libre analytics platform
+ *
+ * @link https://matomo.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+/**
+ * Add or replace a URL parameter
+ * @param url
+ * @param paramName
+ * @param paramValue
+ */
+function addParameterToUrl(url: string, paramName:string, paramValue:string): string {
+  let returnUrl = url;
+  /* eslint-disable prefer-template */
+  const rx = new RegExp('\\b(' + paramName + '=).*?(&|#|$)');
+
+  // Replace any existing parameter
+  if (returnUrl.search(rx) >= 0) {
+    /* eslint-disable prefer-template */
+    return returnUrl.replace(rx, '$1' + paramValue + '$2');
+  }
+  // Add new parameter
+  returnUrl = returnUrl.replace(/[?#]$/, '');
+  /* eslint-disable prefer-template */
+  return returnUrl + (returnUrl.indexOf('?') > 0 ? '&' : '?') + paramName + '=' + paramValue;
+}
+
+/**
+ * Takes a raw URL and returns an HTML link tag for the URL, if the URL is for a matomo.org
+ * domain then the URL will be modified to include campaign parameters
+ *
+ * @param url              URL to
+ * @param campaignOverride Optional
+ * @param sourceOverride   Optional
+ * @param mediumOverride   Optional
+ */
+export function externalRawLink(
+  url: string,
+  campaignOverride: string,
+  sourceOverride: string,
+  mediumOverride: string,
+): string {
+  if (!url) {
+    return '';
+  }
+
+  // Check if matomo.org domain
```

Comment:
Could we simplify the code to work on an URL object?

```javascript
const returnURL = new URL(url);
const validDomain = returnURL.host.endsWith('matomo.org');

if (validDomain && module && ...) {
  ...
  returnUrl.searchParams.set('mtm_campaign', campaign);
  ...
}

return returnUrl.toString();
```

That should result in the same result but we could remove the Regex work done to set the parameters.

Reply:
Thanks @mneudert, that's a much cleaner approach than the custom regex function :+1: